pybop/models/base_model.py

```diff
@@ -197,6 +198,9 @@ def build(
             self._solver._model_set_up = {}  # noqa: SLF001
 
         self.n_states = self._built_model.len_rhs_and_alg  # len_rhs + len_alg
+        self.store_events()
+        if self.allow_infeasible_solutions:
```

Comment:
I don't think "allow_infeasible_solutions" is really an accurate name, surely you mean "turn_off_events", since this is what you do?